src/Microsoft.AspNetCore.StaticFiles/ResponseCacheFilter.cs

```diff
@@ -0,0 +1,98 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Net.Http.Headers;
+
+namespace Microsoft.AspNetCore.StaticFiles
+{
+    internal class ResponseCacheFilter : IResponseCacheFilter
+    {
+        /// <summary>
+        /// Apply the given cacheProfile to the <see cref="HttpContext"/>
+        /// </summary>
+        /// <param name="context">The current <see cref="HttpContext"/></param>
+        /// <param name="cacheProfile">The profile which contains the cache control settings</param>
+        public void ApplyCacheProfile(HttpContext context, CacheProfile cacheProfile)
+        {
+            if (context == null)
+            {
+                throw new ArgumentNullException(nameof(context));
+            }
+            if (cacheProfile == null)
+            {
+                throw new ArgumentNullException(nameof(cacheProfile));
+            }
+
+            var noStore = cacheProfile.NoStore ?? false;
```

Comment:
Although an argument can be made for the `Duration` field, I don't think the same can be said for the `bool` fields. Those can have default values since we assign them here anyway. Now, I also think we should set a default value for duration to something like 10s instead of 0 and skip the check for explicitly setting duration. What do you think @Tratcher?